### ISSUE TITLE
AzureAD User cant start the Service

### ISSUE THREAD

**Issue by fladdie** [DuoStream/Duo#127]
The service does not start on a PC managed by AzureAD because DUO cannot "find" the registered AzureAD user. If I create a local user, everything works. It would be great if AzureAD users were supported.

The only error shown in the event log is:

_The user couldn't be added to the Remote Desktop Users group_

clear, because it does not find any user. In the Duo Manager in the Session Settings there is no user listed in the scroll down.

I am currently using version 1.4.5+

**Comment by Black-Seraph** [comment]
Yeah, that's a known limitation we really can't do all that much about because we're using local accounts + NLA to authenticate at the moment.

We'd have to drop that and find a better way to do authentication, or, even better, give users the option to change the authentication level.

For now all I can suggest is create a local account you can use as a jump-in-point, nothing stops you from running applications as another user from within the local session after all.

**Comment by fladdie** [comment]
Hello, 

I almost figured as much. 
It works with a local user. It would of course be great if Azure AD user authentication is also possible at some point in the future.

Many thanks for your great work.

**Comment by Black-Seraph** [comment]
While it is technically possible to authenticate Duo instances via Kerberos, it just isn't safe to fall back on Kerberos as a default yet due to its centralized nature and it being essentially useless in Workgroup environments.

This might change once Microsoft starts pushing local Kerberos authentication nodes for Home-Edition operating systems, but even then its still not safe to fully kick NTLM to the curb as they already announced that these local authentication nodes will be a Windows 11 only thing, and we can't just leave the Windows 10 users stranded.

This is why Duo uses NTLM / NLA for authentication so far.

I can imagine a transparent fall-through implementation though, something that tries NTLM first, and moves onto Kerberos if NTLM fails.

This will require a few changes in Duo to accomodate the addition of a domain-name parameter though.

I will see what I can do about this and hit you up when a test build is ready.
